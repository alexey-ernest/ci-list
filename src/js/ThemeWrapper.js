/**
 * Wraps component with theme provider to avoid Material UI errors.
 */

import React, {Component, PropTypes} from 'react';

import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
import getMuiTheme from 'material-ui/styles/getMuiTheme';
import {
  lightBlue500,
  lightBlue700,
  grey400,
  pinkA200,
  grey200,
  grey500,
  darkBlack,
  white,
  grey300,
  fullBlack
} from 'material-ui/styles/colors';
import {fade} from 'material-ui/utils/colorManipulator';


// More on Colors: http://www.material-ui.com/#/customization/colors
var muiTheme = getMuiTheme({
  palette: {
    primary1Color: lightBlue500,
    primary2Color: lightBlue700,
    primary3Color: grey400,
    accent1Color: pinkA200,
    accent2Color: grey300,
    accent3Color: grey500,
    textColor: darkBlack,
    alternateTextColor: white,
    canvasColor: white,
    borderColor: grey300,
    disabledColor: fade(darkBlack, 0.3),
    pickerHeaderColor: lightBlue500,
    clockCircleColor: fade(darkBlack, 0.07),
    shadowColor: fullBlack,
  }
});

export default class ThemeWrapper extends Component {

  static childContextTypes = {
    muiTheme: React.PropTypes.object.isRequired
  };

  getChildContext() {
    return {muiTheme: muiTheme};
  }

  render() {
    return (
      <MuiThemeProvider muiTheme={muiTheme}>
        {this.props.children}
      </MuiThemeProvider>
    );
  }
}
