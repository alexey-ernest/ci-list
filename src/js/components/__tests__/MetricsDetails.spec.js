
import React from 'react';
import renderer from 'react-test-renderer';

jest.mock('react-dom');

jest.mock('material-ui/CircularProgress', () => {
  return () => {
    return (
      <p>CircularProgress</p>
    );
  }
});

jest.mock('../../actions/JobActionCreators', () => {
  return {
    requestJobMetrics: jest.fn()
  };
});

jest.mock('react-chartjs', () => {
  return {
    Line: () => {
      return (
        <div>Chartjs</div>
      );
    }
  }
});

import ThemeWrapper from '../../ThemeWrapper';
import MetricsDetails from '../MetricsDetails.react';


describe('MetricsDetails component', () => {

  it('should render correctly when there is no tests data', () => {
    const component = renderer.create(
      <ThemeWrapper>
        <MetricsDetails job={{}} />
      </ThemeWrapper>
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should request data for the job', () => {
    const actionCreators = require('../../actions/JobActionCreators');
    const spy = jest.fn();
    actionCreators.requestJobMetrics = spy;

    const job = {
      id: 5
    };

    const component = renderer.create(
      <ThemeWrapper>
        <MetricsDetails job={job} />
      </ThemeWrapper>
    );

    expect(spy.mock.calls.length).toBe(1);
    expect(spy.mock.calls[0][0]).toBe(job.id);
  });

  it('should render chart when tests property specified', () => {
    const metrics = {
      datasets: [{
        data: [1, 2, 3]
      }]
    };
    const component = renderer.create(
      <ThemeWrapper>
        <MetricsDetails job={{}} metrics={metrics} />
      </ThemeWrapper>
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

});
