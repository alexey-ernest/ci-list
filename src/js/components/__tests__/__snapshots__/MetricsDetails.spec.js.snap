exports[`MetricsDetails component should render chart when tests property specified 1`] = `
<div>
  <div>
    Chartjs
  </div>
</div>
`;

exports[`MetricsDetails component should render correctly when there is no tests data 1`] = `
<div>
  <p>
    CircularProgress
  </p>
</div>
`;
